protos/flyteidl/service/admin.proto

```diff
@@ -65,18 +76,18 @@ service AdminService {
       body: "*"
     };
     option (grpc.gateway.protoc_gen_swagger.options.openapiv2_operation) = {
-      description: "Create and register a workflow definition"
-      responses: {
-        key: "400"
-        value: {
-          description: "Returned for bad request that may have failed validation."
-        }
-      }
-      responses: {
-        key: "409"
-        value: {
-          description: "Returned for a request that references an identical entity that has already been registered."
-        }
+      description: "Create and register a workflow definition."
+      responses: {	
+        key: "400"	
+        value: {	
+          description: "Returned for bad request that may have failed validation."	
+        }	
+      }	
+      responses: {	
+        key: "409"	
+        value: {	
+          description: "Returned for a request that references an identical entity that has already been registered."	
```

Comment:
I don't think this is really descriptive of our idempotent behavior. It isn't returned for an identical entity, it is returned for an entity that has already been registered with the same ID _and_ which is not identical to the original entry.

Reply:
the translation from grpc status code to http is confusing, but this is in fact what we return when the digests match (https://github.com/grpc-ecosystem/grpc-gateway/blob/master/runtime/errors.go#L31). we return InvalidArgument otherwise but there's a lot of scenarios where InvalidArgument is the response so i don't think calling all of them out is particularly useful since they're likely to evolve

Reply:
ahh i see